### PR TITLE
Don't fail silently if trying to add a collaborator twice

### PR DESCRIPTION
**Pull request by adelowo**
While wanting to work on #4507, I noticed that adding a collaborator twice leads to a ___success message___ saying **collaborator has been added** when the collaborator wasn't infact added again, I think that is the way xorm's `InsertOne` works ( correct me if wrong).. This PR makes sure to check if the user is already a collaborator thus returning early ....

### CODE REVIEW

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -1014,6 +1014,7 @@ settings.transfer_succeed = The repository has been transferred.
 settings.confirm_delete = Delete Repository
 settings.add_collaborator = Add Collaborator
 settings.add_collaborator_success = The collaborator has been added.
+settings.add_collaborator_duplicate = The collaborator has been previously added.
```
Comment (lafriks) [inline options/locale/locale_en-US.ini]: Message should be: `The collaborator is already added to this repository.`

**Comment by techknowlogick** [comment]
Could you add an integration test?

(sorry about the accidental close)

**Comment by adelowo** [comment]
~~Sure... Would add those in a bit~~

Done @techknowlogick

**Review by techknowlogick — APPROVED** [review]

**Comment by lunny** [comment]
LGTM

**Pull request merged**